.travis.yml

```diff
@@ -19,8 +16,14 @@ matrix:
       python: 3.8
     - arch: amd64
       python: pypy
+      before_install:
+        # Workaround for dependency errors, probably due to https://github.com/z4r/python-coveralls/issues/66
```

Comment:
Hmm, why don't require a specific version in the setup.cfg or, perhaps better just blacklist wrong versions?

Reply:
I have no clue why the problem only exists for pypy but not for CPython. What do you suggest?

Reply:
I don't know.  Perhaps, because PyPy != CPython in general, there are some incompatibilities (=PyPy bugs).

> What do you suggest?

You can blacklist "wrong" pytest-cov versions in the setup.cfg, using operators (!= or <).   See e.g. [this](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#declaring-required-dependency).
I think, you can even restrict this version restriction _only_ for PyPy (AFAIR this is something like: "pytest-cov != x.y.z; implementation_name='pypy'", you could take look on PEP 508 and setuptools docs).

Reply:
I now found the actual cause, which is a bug in pip that occurs if you an old version of an indirect dependency is already installed. This means the workaround is no longer needed.

Reply:
Hmm, maybe you could just set minimal pytest requirements in the setup.cfg.  This looks to me as a simplest solution.

"pytest >= 4.6" seems to be working, see [build logs](https://travis-ci.org/github/skirpichev/mpmath/builds/732148303).  Everything else then should be reverted (this works for all versions in the master) and, perhaps, you will want also rebase the PR to squash all commits to single one.